src/client/storage/index.js

```diff
@@ -75,20 +75,14 @@ const getChunk = async (chunkId, encoding = 'utf8', useCache = true) => {
     // chunks from the bundler's backup (S3).
     try {
         log.debug({chunkId}, 'Downloading chunk from bundler backup');
-        const {data} = await axios.get(`${BUNDLER_DOWNLOAD_URL}/${chunkId}`);
+        const {data: buf} = await axios.request({
+            method: 'GET',
+            url: `${BUNDLER_DOWNLOAD_URL}/${chunkId}`,
+            responseType: 'arraybuffer'
+        });
         log.debug({chunkId}, 'Successfully downloaded chunk from Bundler backup');
-        const buf =
-            typeof data === 'object' ? Buffer.from(JSON.stringify(data)) : Buffer.from(data);
         await fs.writeFile(chunkPath, buf);
         chunk.size = buf.length;
-
-        const hash = hashFn(buf).toString('hex');
-        if (hash !== chunk.id) {
-            const errMsg = 'Chunk id and data do not match';
-            log.warn({chunkId: chunk.id, hash}, errMsg);
-            throw new Error(errMsg);
-        }
```

Comment:
In order to make this handler a bit quicker and simpler, I think it's safe to skip this validation (lines 85:90) for content that comes from S3 since only _Bundler_ has write access to the bucket. We don't have the same risk as we do with Arweave, where anyone can upload random content and "fake" our tags.

Reply:
I think that's ok having in mind that in my latest change to the bundler I'm validating the content on s3. On the other hand thanks to this part of the code we become aware about that something was wrong. I'm not 100%.